### ISSUE TITLE
Are we ready for merge?

### ISSUE THREAD

**Issue by tailhook** [tailhook/vagga-docker#1]
The issue number one should track the status of the project and whether it's ready to be merged into mainstream vagga.

Currently, I have the following todo list (but it's not comprehensive, will add more later):
- [x] `_expose-ports` option
- [ ] `docker create network vagga`, so that connections between vagga processes on localhost work
- [ ] fix expose ports combined with `--only` and `--exclude`
- [ ] support for settings in `settings.yaml` (mirrors, external-volumes...) (#4)
  - [x] ubuntu/alpine mirror
  - [ ] `external-volumes`
- [ ] auto-add `pass-tcp-socket` to `_expose-ports`
- [ ] expose ports for `_run`
- [ ] `_expose-dirs` option (unlikely to be implemented before merge, because requires tweaks in the build process)
- [ ] completion (also not critical so will likely happen after merge)

The issues so far:
- Ports must be bound on `0.0.0.0` not `127.0.0.1` as we usually do. This problem also exists in vagrant. Is there any way to fix it? Make our own port proxy?
- Container start and shutdown takes about 2 seconds on my linux. Similar command on native vagga runs less than 100 ms. Unfortunately overhead of python is at least 0.7 s. Python overhead will be removed after the merge.

The questions so far before we can proceed:
- Is python3.4 fine for the prototype, or is it hard to install on OS X?
- Does performance acceptable? (to measure python overhead run `vagga --help`)

/cc @vmagamedov, @zemlanin, @AlexMost

**Comment by zemlanin** [comment]
Python 3.4 is fine (it is installable with single `brew install python3` command and `brew` is a must for mac development)

**Comment by evacchi** [comment]
hey, just learned about this project, cool effort!

I wanted to share my two cents. Alpine it has proven a bit risky for us because it ships with "nonstandard" libc thus sometimes precompiled binaries might not work. e.g., we had problem with pre-built `protoc` (https://github.com/os72/protoc-jar) that we use for repeatable builds in our projects, because it expects glibc.

BTW, I'll try a few examples and let you know if I encounter problems

**Comment by tailhook** [comment]
@evacchi, I don't think it matters. You can use ubuntu as guest system regardless of what we use for virtual machine wrapping code. The guest system is where your code runs in.

But let us know if there are any problems anyway.
